test/e2e/persistent_volumes.go

```diff
@@ -738,6 +739,109 @@ var _ = framework.KubeDescribe("PersistentVolumes", func() {
 			Expect(err).NotTo(HaveOccurred())
 		})
 	})
+
+	///////////////////////////////////////////////////////////////////////
+	//				Vsphere Disk
+	///////////////////////////////////////////////////////////////////////
+	// Testing configurations of single a PV/PVC pair attached to a Vsphere Disk
+
+	framework.KubeDescribe("PersistentVolumes:vsphere", func() {
+
+		var (
+			volumePath string
+			pv        *v1.PersistentVolume
+			pvc       *v1.PersistentVolumeClaim
+			clientPod *v1.Pod
+			pvConfig  persistentVolumeConfig
+		)
+		BeforeEach(func() {
+			framework.SkipUnlessProviderIs("vsphere")
+			if volumePath == "" {
+				vsp, err := vsphere.GetVSphere()
+				Expect(err).NotTo(HaveOccurred())
+				var volumeoptions vsphere.VolumeOptions
+				volumeoptions.CapacityKB=1 * 1024 * 1024
+				volumeoptions.Name="e2e-disk" + time.Now().Format("20060102150405")
+				volumeoptions.DiskFormat="thin"
+
+				volumePath, err = vsp.CreateVolume(&volumeoptions)
+				Expect(err).NotTo(HaveOccurred())
+				pvConfig = persistentVolumeConfig{
+					namePrefix: "vspherepv-",
+					pvSource: v1.PersistentVolumeSource{
+						VsphereVolume: &v1.VsphereVirtualDiskVolumeSource{
+							VolumePath:	volumePath,
+							FSType:		"ext4",
+						},
+					},
+					prebind: nil,
+				}
+			}
+		})
+
+		AfterEach(func() {
+			framework.Logf("AfterEach: Cleaning up test resources")
+			if c != nil {
+				deletePod(f, c, ns, clientPod)
+				pvPvcCleanup(c, ns, pv, pvc)
+				clientPod = nil
+				pvc = nil
+				pv = nil
+			}
+		})
+
+		AddCleanupAction(func() {
+			if len(volumePath) > 0 {
+				vsp, err := vsphere.GetVSphere()
+				Expect(err).NotTo(HaveOccurred())
+				vsp.DeleteVolume(volumePath)
+			}
+		})
+
+		// Attach a persistent disk to a pod using a PVC.
+		// Delete the PVC and then the pod.  Expect the pod to succeed in unmounting and detaching PD on delete.
```

Comment:
We delete the pvc before the pod here.. if the reclaim policy needs deletion of volume what do we expect?